cyral/data_source_cyral_integration_idp.go

```diff
@@ -0,0 +1,167 @@
+						},
+						"single_sign_on_service_url": {
+							Description: "Single sign on service URL.",
+							Type:        schema.TypeString,
+							Computed:    true,
+						},
+						"enabled": {
+							Description: "True if the IdP integration is enabled.",
+							Type:        schema.TypeBool,
+							Computed:    true,
+						},
+					},
+				},
+				Computed:    true,
+				Description: "Set of existing IdP integrations for the given filter criteria.",
+			},
+		},
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+	}
+}
+
+func dataSourceIntegrationIdPRead(
+	ctx context.Context,
+	d *schema.ResourceData,
+	m interface{},
+) diag.Diagnostics {
+	log.Printf("[DEBUG] Init dataSourceIntegrationIdPRead")
+	c := m.(*client.Client)
+
+	idpTypeFilter := d.Get("type").(string)
+
+	var url = fmt.Sprintf("https://%s/v1/integrations/saml", c.ControlPlane)
+	if idpTypeFilter != "" {
+		url = fmt.Sprintf("https://%s/v1/integrations/saml?identityProvider=%s",
+			c.ControlPlane, idpTypeFilter)
+	}
+
+	body, err := c.DoRequest(url, http.MethodGet, nil)
+	if err != nil {
+		return createError("Unable to execute request to read idp integrations", err.Error())
+	}
+
+	var idpIntegrations = IdPIntegrations{}
+	if err := json.Unmarshal(body, &idpIntegrations); err != nil {
+		return createError("Unable to unmarshal idp integrations", err.Error())
+	}
+	log.Printf("[DEBUG] Response body (unmarshalled): %#v", idpIntegrations)
+
+	var idpList []interface{}
+	displayNameFilter := d.Get("display_name").(string)
+
+	log.Printf("[DEBUG] display_name: %s", displayNameFilter)
+	log.Printf("[DEBUG] type: %s", idpTypeFilter)
+	if idpIntegrations.Connections != nil {
+		for _, connection := range idpIntegrations.Connections.Connections {
+			log.Printf("[DEBUG] Connection: %#v", connection)
+			if connection != nil {
+				// Skip in case filters are non-empty but
+				if (displayNameFilter != "" && displayNameFilter != connection.DisplayName) ||
+					(idpTypeFilter != "" && !strings.HasPrefix(connection.Alias, idpTypeFilter)) {
+					continue
+				}
+				// Conditions to return data:
+				// 1. displayNameFilter is not empty and correspond to the one in the current connection;
+				// 2. displayNameFilter and idpTypeFilter are empty.
+				// 3. idpTypeFilter is not empty (already filtered in the endpoint call)
+				// Conditions NOT to return data:
+				// 4. displayNameFilter is not empty and does not correspond to the one in the current connection OR
+				//    idpTypeFilter is not empty and does not correspond to the one in the current connection
+				if ((displayNameFilter != "" && displayNameFilter == connection.DisplayName) || // 1
+					(displayNameFilter == "" && idpTypeFilter == "") || // 2
+					(idpTypeFilter != "")) && // 3
+					!((displayNameFilter != "" && displayNameFilter != connection.DisplayName) || // 4
+						(idpTypeFilter != "" && !strings.HasPrefix(connection.Alias, idpTypeFilter))) {
```

Comment:
Why do we need these conditionals? The API will already filter by type, then we just need to worry if it needs to be filtered by display_name, but we already check that in the previous `if(displayNameFilter != "" && displayNameFilter != connection.DisplayName)`.

If we reach this part of the code, the idp is already filtered by type (in the API call) and by `display_name`, since it passed the `if` mentioned above. Also, if no filter is passed, then the API will not filter by type and the ìf`above will not filter by `display_name` also. I think we can just remove these conditionals and just append the item to the `idpList`
```suggestion
```

Reply:
Indeed. I removed the unused code. Thanks.